interop/tapir/src/main/scala/caliban/interop/tapir/HttpInterpreter.scala

```diff
@@ -214,4 +214,24 @@ object HttpInterpreter {
 
     postEndpoint :: getEndpoint :: Nil
   }
+
+  /**
+   * Creates an endpoint that serves the GraphiQL UI from CDN.
+   *
+   * @param apiPath The path at which the API can be introspected.
+   *
+   * @see [[https://github.com/graphql/graphiql/tree/main/examples/graphiql-cdn]]
+   */
+  def makeGraphiqlEndpoint(apiPath: String): Endpoint[Unit, Right[Nothing, String], Nothing, String, Any] = {
+    val apiPath0 = apiPath.split("/").filter(_.nonEmpty).mkString("/", "/", "")
+    infallibleEndpoint.get
+      .in(extractFromRequest(_.pathSegments))
+      .out(htmlBodyUtf8)
+      .mapIn { segments =>
+        val uiPath = segments.mkString("/", "/", "")
+        Right(HttpUtils.graphiqlHtml(apiPath = apiPath0, uiPath = uiPath))
+      }(
+        _.value.split("/").filter(_.nonEmpty).toList
+      )
```

Comment:
Does anyone think of a better way to do this? It works but I'm not too happy with this

Reply:
When is this even used?

Reply:
I mean a less clean way but more efficient way would be to combine the steps into a single operation. But if it's not a hot path I wouldn't worry about it

Reply:
I ended up reimplementing the endpoint properly, requiring an implicit MonadError (from sttp) to create it. It should be possible now to customize the endpoint using the `.endpoint` method on the `ServerEndpoint.Full`